### DISCUSSION TITLE
Thonny Micropython and Unknown IR Remote

### DISCUSSION THREAD

**Discussion by SeregaZ2004** [micropython/micropython#11650] [RP2040 / Pico] [answered]
Hello. I have zero experience with Raspberry Pico. I have 38kHz IR Reciver module and this library:
https://github.com/peterhinch/micropython_ir/tree/master

When i try to test IR by:
from ir_rx.test import test

and test() for all types. They always show errors:
Invalid start pulse (test(6))
Error: bad block (test(2))
Error: bad block and Invalid start pulse (for test(4)
Error: overrun (test(5))

Maybe some one know how to get data from this IR remote? Probably this remote have mistake - seller write 433mhz. But it is radio frequency, not IR. But this remote definetly IR - USB reciever have this "lamp" to signal detection. And if i cover that reciever lamp - it stop working. So it is sure not a radio. And my 38kHz module for Piko flashing light, when i press something on a this remote. Other remote for TV work fine with test() or test(1) - NEC type. but this is one - USB PC - cursed. Any way to fix that problem?

**Comment by davefes** [comment]
Is that 38kHz not 38MHz?

**Reply by SeregaZ2004** [reply]
my mistake. 38kHz.

**Reply by davefes** [reply]
OK, can you post a link to the actual device in question?

**Reply by SeregaZ2004** [reply]
i am buy from this seller: https://aliexpress.ru/item/1005003177507391.html

**Reply by davefes** [reply]
The only thing I could read was "Wireless Communication IR".  Had a quick read through https://github.com/peterhinch/micropython_ir/tree/master
but no experience using it.

Good luck!

**Comment by SeregaZ2004** [comment]
Some magic glitch happens with my Raspberry Piko. It live own live :) Test code can work, can not work by own wish. Somehow it start work and i have this data:
`
nm\bt		1but		1but		1but				2but		2but				3but
0		278		249		269				278		251				272
1		1643		1695		1647				1643		1664				1646
2		338		310		326				335		301				330
3		1677		1675		1665				1650		1690				1656
4		313		312		296				330		299				329
5		1652		1656		1691				1656		1690				1661
6		335		331		300				335		296				328
7		714		696		728				719		731				724
8		338		333		296				313		299				309
9		689		717		733				715		725				694
10		316		311		297				336		304				326
11		713		715		732				692		726				724
12		343		314		296				312		300				312
13		1644		1673		1691				1651		1690				1653
14		345		315		324				332		324				333
15		682		716		703				721		706				720
16		345		314		302				335		295				313
17		1644		1673		1688				1628		1691				1651
18		347		290		299				360		298				333
19		1623		1700		1690				1630		1689				1656
20		363		312		287				331		302				331
21		1645		1675		1703				1658		1691				1656
22		343		316		297				331		299				331
23		1648		1676		1689				719		728				1659
24		341		314		302				337		297				356
25		1625		1674		1689				1627		1691				698
26		363		317		322				333		304				336
27		687		714		704				722		724				690
28		341		313		272				339		300				337
29		688		711		757				687		727				689
30		339		316		324				337		327				334
31		687		715		705				689		704				695
32		342		313		297				341		296				339
`

First 3 is one button, then 2 - second button. and last column - third button. As i can understand this values need to be viewed as 1 and 0. Somekind of:
300 + - = 0
1600 + - = 1
but here is something another 700 + - middle between 300 and 1600. maybe 700 it is two 300...

So i have two questions - how to apply this data to main code. And can all this library have some kind of puzzle? I mean i no need some NEC, Phillips or other code. Only this raw reading and recognize. Some small part of that - to save memory. How to do this cut operation for main code?

**Reply by SeregaZ2004** [reply]
I decide to count values this way:
Case 200 To 400 as pause - *
Case 500 To 900 as 0
Case 1400 To 1900 as 1

So table start looks like this:
`
A	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	1	*	1	*	0	*	0	*	0	*	
A	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	1	*	1	*	0	*	0	*	0	*	
A	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	1	*	1	*	0	*	0	*	0	*	
B	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	0	*	1	*	0	*	0	*	0	*	
B	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	0	*	1	*	0	*	0	*	0	*	
C	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	1	*	1	*	0	*	0	*	0	*	0	*	
D	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	0	*	
E	*	1	*	1	*	1	*	0	*	0	*	0	*	1	*	0	*	0	*	0	*	0	*	0	*	0	*	1	*	0	*	0	*	


 `

**Comment by SeregaZ2004** [comment]
This is a super video with explanations... but where to get that pulseio to recreate that he talking? It is not a Micropython for Raspberry Pico?
https://www.youtube.com/watch?v=TIbp7DzfOBM

**Comment by SeregaZ2004** [comment, marked as answer]
Solve. I take example of that person:
https://techtotinker.com/2021/08/044-micropython-technotes-infrared-receiver/
He take only that is needed for save memory - not all library. Then i try to kick out some items, that i am no need and add new checks for millisec values for my protocol. It is a dirty code, because i have no idea how to delete some part of code. When i delete it - code shows errors. Maybe later, when i will more advanced - will remove it.

This code have already array, that it get from IR Module. So i just need to read values, and comparise to reference.
